Makefile

```diff
@@ -14,7 +14,7 @@ install-pip-setuptools:
 	python -m pip install -U pip setuptools wheel
 
 lint:
-	cd $(plugin) && pre-commit run -a --hook-stage manual
+	pre-commit run -a --hook-stage manual
```

Comment:
Probably being naive again, but isn't the problem of conflicting dependencies that doing `run -a` (instead of only doing `pylint-$(plugin)` from that set)? The way things are currently set up.

Reply:
More broadly, we have the general problem that monorepo support isn't great (existent) in pre-commit. Could consider a next step of trying ruff for just pylint (and then extending it more broadly)? I guess @noklam could have some thoughts based on his work on the main repo.

Reply:
> Probably being naive again, but isn't the problem of conflicting dependencies that doing `run -a` (instead of only doing `pylint-$(plugin)` from that set)? The way things are currently set up.

I think this would be the best solution for now and we should look into ruff for kedro-plugins.

Reply:
`ruff` definitely helps for `isort` ,`flake8` and `pylint`, many of these pre-commit libraries are pinned at an old version, so when I switch to `ruff` I get lots of error.

Reply:
I've updated `make lint` it will now run each individual pre-commit hook instead of doing it in `check-plugin.yml so we can continue to use `make plugin=${{ inputs.plugin }} lint` in `check-plugins.yml`.

The new `make lint`:

```
lint:
	pre-commit run trailing-whitespace --all-files && pre-commit run end-of-file-fixer --all-files && pre-commit run check-yaml --all-files && pre-commit run check-added-large-files --all-files && pre-commit run check-case-conflict --all-files && pre-commit run check-merge-conflict --all-files && pre-commit run debug-statements --all-files && pre-commit run requirements-txt-fixer --all-files && pre-commit run flake8 --all-files && pre-commit run isort-$(plugin) --all-files --hook-stage manual && pre-commit run black-$(plugin) --all-files --hook-stage manual && pre-commit run secret_scan --all-files --hook-stage manual && pre-commit run bandit --all-files --hook-stage manual && pre-commit run pylint-$(plugin) --all-files --hook-stage manual && pre-commit run pylint-$(plugin)-features --all-files --hook-stage manual && pre-commit run pylint-$(plugin)-tests --all-files --hook-stage manual
```